sky/engine/core/dart/text.dart

```diff
@@ -65,18 +90,66 @@ enum TextBaseline {
 }
 
 /// A linear decoration to draw near the text
-enum TextDecoration {
+class TextDecoration {
+  const TextDecoration._(this._mask);
+
+  /// Constructs a decoration that paints the union of all the given decorations.
+  TextDecoration.combine(List<TextDecoration> decorations) {
+    _mask = 0;
+    for (TextDecoration decoration in decorations)
+      _mask |= decoration._mask;
+  }
+
+  final int _mask;
+
+  /// Whether this decoration will paint at least as much decoration as the given decoration.
+  bool contains(TextDecoration other) {
+    return (_mask | other._mask) == _mask;
+  }
+
   /// Do not draw a decoration
```

Comment:
maybe use hex constants here? for some reason I associate those more with bit patterns than decimals. Maybe that's just me though.